src/domManipulation.ts

```diff
@@ -4,16 +4,27 @@ import fetchWeatherForecast from "./api";
 
 // Get the parent element for the main cities/locations which holds all the buttons.
 const mainCities = document.getElementById("main-cities-nav");
+
 // Get all the button elements in this parent.
-const mainCityButtons = mainCities.getElementsByClassName("city-chip");
+let  mainCityButtons : HTMLCollectionOf<Element>;
+if (mainCities !== null) {
+  mainCityButtons = mainCities.getElementsByClassName("city-chip");
+}
+
 // Set the default city to the first option.
 let selectedMainCity = 0;
 
 // Toggle visibility of the pop-up
 const mainLocations = document.getElementById("main-cities-container");
 const customLocationPopup = document.getElementById("popup-container");
-mainLocations.style.visibility = "visible";
-customLocationPopup.style.visibility = "hidden";
+if (mainLocations !== null) {
+  mainLocations.style.visibility = "visible";
+}
+if (customLocationPopup !== null)
```

Comment:
Simplify to `if (customLocationPopup)`

Reply:
This has been resolved in a previous comment, the code has been removed entirely as mentioned above.